airflow/jobs/local_task_job.py

```diff
@@ -151,16 +147,16 @@ def signal_handler(signum, frame):
             self.on_kill()
 
     def handle_task_exit(self, return_code: int) -> None:
-        """Handle case where self.task_runner exits by itself"""
+        """Handle case where self.task_runner exits by itself or received a SIGKILL"""
```

Comment:
SIGKILL can't be caught though.

>SIGKILL is an immediate termination signal and it cannot be caught or ignored by the process.

Reply:
The wording is pretty inaccurate here (inherited from previous implemented, see `on_kill` and line 161 below); `handle_task_exit()` is reached when the task is killed, but not necessarily by a SIGKILL. This should proably say something like “or is killed externally”.

Reply:
Yes. @kaxil , actually, it's the task that gets the sigkill, -9 instead, somehow, it gets to the handle task exit. See https://github.com/apache/airflow/issues/11086

Reply:
Yup, just updating docstrings as TP mentioned though